src/game/inventory/inventory.c

```diff
@@ -900,81 +902,27 @@ static int32_t Inv_ConstructAndDisplay(int inv_mode)
-                return GF_STORY_SO_FAR | g_GameInfo.current_save_slot;
-            } else if (g_GameInfo.passport_page == PASSPORT_PAGE_2) {
-                // page 2: restart level
-                return GF_RESTART_GAME | g_CurrentLevel;
-            } else {
-                // page 3: exit game
-                return GF_EXIT_TO_TITLE;
-            }
+        if (g_GameInfo.passport_selection == PASSPORT_MODE_LOAD_GAME) {
+            return GF_START_SAVED_GAME | g_GameInfo.current_save_slot;
+        } else if (
+            g_GameInfo.passport_selection == PASSPORT_MODE_SELECT_LEVEL) {
+            return GF_SELECT_GAME | g_GameInfo.select_level_num;
+        } else if (
+            g_GameInfo.passport_selection == PASSPORT_MODE_STORY_SO_FAR) {
+            return GF_STORY_SO_FAR | g_GameInfo.current_save_slot;
+        } else if (g_GameInfo.passport_selection == PASSPORT_MODE_NEW_GAME) {
+            Savegame_InitCurrentInfo();
+            return GF_START_GAME | g_GameFlow.first_level_num;
+        } else if (g_GameInfo.passport_selection == PASSPORT_MODE_SAVE_GAME) {
+            Savegame_Save(g_GameInfo.current_save_slot, &g_GameInfo);
+            Config_Write();
+            return GF_NOP;
+        } else if (g_GameInfo.passport_selection == PASSPORT_MODE_RESTART) {
+            return GF_RESTART_GAME | g_CurrentLevel;
+        } else if (g_GameInfo.passport_selection == PASSPORT_MODE_EXIT_GAME) {
+            return GF_EXIT_GAME;
         } else {
-            if (g_GameInfo.passport_page == PASSPORT_PAGE_1
-                && g_GameInfo.passport_mode == PASSPORT_MODE_SHOW_SAVES) {
-                // page 1: load game
-                return GF_START_SAVED_GAME | g_GameInfo.current_save_slot;
-            } else if (
-                g_GameInfo.passport_page == PASSPORT_PAGE_1
-                && g_GameInfo.passport_mode == PASSPORT_MODE_SELECT_LEVEL) {
-                // page 1: select level
-                return GF_SELECT_GAME | g_GameInfo.select_level_num;
-            } else if (
-                g_GameInfo.passport_page == PASSPORT_PAGE_1
-                && g_GameInfo.passport_mode == PASSPORT_MODE_STORY_SO_FAR) {
-                // page 1: story so far
-                return GF_STORY_SO_FAR | g_GameInfo.current_save_slot;
-            } else if (g_GameInfo.passport_page == PASSPORT_PAGE_2) {
-                if (g_CurrentLevel == g_GameFlow.gym_level_num) {
-                    // page 2: new game in gym
-                    Savegame_InitCurrentInfo();
-                    return GF_START_GAME | g_GameFlow.first_level_num;
-                } else {
-                    // page 2: save game
-                    Savegame_Save(g_GameInfo.current_save_slot, &g_GameInfo);
-                    Config_Write();
-                    return GF_NOP;
-                }
-            } else {
-                // page 3: exit to title
-                return GF_EXIT_TO_TITLE;
-            }
+            return GF_EXIT_TO_TITLE;
```

Comment:
Maybe we could convert it to a switch? Do you reckon we might need more precise conditions in the future?